sagemaker/launch/launch_realtime_endpoint.py

```diff
@@ -0,0 +1,313 @@
+"""
+    Copyright Contributors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Launch a SageMaker endpoint for realtime inference
+"""
+
+import os
+import argparse
+import logging
+import tempfile
+from time import gmtime, strftime
+
+import boto3 # pylint: disable=import-error
+from botocore.exceptions import WaiterError
+import sagemaker as sm
+
+from launch_utils import (wrap_model_artifacts,
+                          check_tarfile_s3_object,
+                          upload_data_to_s3,
+                          check_name_format)
+
+
+# TODO: When adding new realtime inference tasks, modify this list
+SUPPORTED_REALTIME_INFER_NC_TASK = 'node_classification'
+SUPPORTED_REALTIME_INFER_TASKS = [SUPPORTED_REALTIME_INFER_NC_TASK]
+
+# Constants for SageMaker endpoints
+ENTRY_FOLDER_NAME = 'realtime_entry_points'
+# TODO: When add new realtime inference tasks, modify this list
+ENTRY_FILE_NAMES = [
+    'node_classification_entry.py',
+    ]
+DEFAULT_GS_MODLE_FILE_NAME = 'model.pt'
+
+
+def run_job(input_args):
+    """ The procedure of deploying a SageMaker real-time inference endpoint
+    
+    Following SageMaker's document for deploying model for real-time inference in
+    https://docs.aws.amazon.com/sagemaker/latest/dg/realtime-endpoints-deploy-models.html,
+    The general steps of a deployment include:
+    1. prepare model artifacts;
+    2. create a deployable model;
+    3. create an endpoint configuration based on the deployable model;
+    4. create an endpoint based on the endpoint configuration.
+
+    This job follows these steps.
+
+    Parameters:
+    -----------
+    region: str
+        The AWS region where the SageMaker endpoint will be deployed.
+    image_url: str
+        The URL of a GraphStorm SageMaker real-time inference Docker image that is located at
+        Amazon ECR in the same region specified in the `region` argument.
```

Comment:
AWS terminology uses URI (Unique Resource Identifier) when referring to ECR images, see https://docs.aws.amazon.com/AmazonECR/latest/userguide/image-info.html

Let's change all references from `url` to be `uri`


```suggestion
    image_uri: str
        The URI of a GraphStorm SageMaker real-time inference Docker image that is located at
        Amazon ECR in the same region specified in the `region` argument.
```